extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -143,6 +143,22 @@ public MSQControllerTask(
     addToContext(Tasks.FORCE_TIME_CHUNK_LOCK_KEY, true);
   }
 
+  public MSQControllerTask(
+      @Nullable String id,
+      MSQSpec querySpec,
+      @Nullable String sqlQuery,
+      @Nullable Map<String, Object> sqlQueryContext,
+      @Nullable SqlResults.Context sqlResultsContext,
+      @Nullable List<SqlTypeName> sqlTypeNames,
+      @Nullable List<ColumnType> nativeTypeNames,
+      @Nullable Map<String, Object> context,
+      Injector injector
```

Comment:
When do we require a custom injector to be supplied? I see it is called in the `MSQCompactionRunner` and we are passing it the same injector there (annotated with `@JacksonInject`).

Reply:
`MSQControllerTask` itself task gets its injector from the json deserializer in the usual task submission scenario. In `MSQCompactionRunner`, since we are creating the instance using the constructor instead, the injector doesn't get instantiated. So we resort to creating the injector while deserializing `MSQCompactionRunner` and then passing it along.